lib/puppet/file_system/memory_impl.rb

```diff
@@ -39,7 +39,7 @@ def path_string(object)
     object.path
   end
 
-  def read(path)
+  def read(path, opts = {})
     handle = assert_path(path).handle
     handle.read
```

Comment:
Interestingly, [StringIO#read](http://ruby-doc.org/stdlib-1.9.3/libdoc/stringio/rdoc/StringIO.html#method-i-read) doesn't take a hash like the other `read` implementations, even though it's also built on `IO` - https://github.com/ruby/ruby/blob/trunk/ext/stringio/stringio.c#L1276-L1337